conformance/utils/config/timeout.go

```diff
@@ -0,0 +1,129 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import "time"
+
+type TimeoutConfig struct {
+	// CreateTimeout represents the maximum time for a Kubernetes object to be created.
+	// Max value for conformant implementation: None
+	// Used by: MustApplyObjectsWithCleanup, MustApplyWithCleanup
+	CreateTimeout time.Duration
+
+	// DeleteTimeout represents the maximum time for a Kubernetes object to be deleted.
+	// Max value for conformant implementation: None
+	// Used by: MustApplyObjectsWithCleanup, MustApplyWithCleanup
+	DeleteTimeout time.Duration
+
+	// GatewayMustHaveAddress represents the maximum time for at least one IP Address has been set in the status of a Gateway.
+	// Max value for conformant implementation: None
+	// Used by: WaitForGatewayAddress
+	GatewayMustHaveAddress time.Duration
+
+	// GatewayStatusMustHaveListeners represents the maximum time for a Gateway to have listeners in status that match the expected listeners.
+	// Max value for conformant implementation: None
+	// Used by: GatewayStatusMustHaveListeners
+	GatewayStatusMustHaveListeners time.Duration
+
+	// GWCMustBeAccepted represents the maximum time for a GatewayClass to have an Accepted condition set to true.
+	// Max value for conformant implementation: None
+	// Used by: GWCMustBeAccepted
+	GWCMustBeAccepted time.Duration
+
+	// HTTPRouteMustNotHaveParents represents the maximum time for an HTTPRoute to have either no parents or a single parent that is not accepted.
+	// Max value for conformant implementation: None
+	// Used by: HTTPRouteMustHaveNoAcceptedParents
+	HTTPRouteMustNotHaveParents time.Duration
+
+	// HTTPRouteMustHaveCondition represents the maximum time for an HTTPRoute to have the supplied Condition.
+	// Max value for conformant implementation: None
+	// Used by: HTTPRouteMustHaveCondition
+	HTTPRouteMustHaveCondition time.Duration
+
+	// HTTPRouteMustHaveParents represents the maximum time for an HTTPRoute to have parents in status that match the expected parents.
+	// Max value for conformant implementation: None
+	// Used by: HTTPRouteMustHaveParents
+	HTTPRouteMustHaveParents time.Duration
+
+	// ManifestFetchTimeout represents the maximum time for getting content from a https:// URL.
+	// Max value for conformant implementation: None
+	// Used by: getContentsFromPathOrURL
+	ManifestFetchTimeout time.Duration
+
+	// NamespacesMustBeReady represents the maximum time for all Pods and Gateways in a namespaces to be marked as ready.
+	// Max value for conformant implementation: None
+	// Used by: NamespacesMustBeReady
+	NamespacesMustBeReady time.Duration
+
+	// RequestTimeout represents the maximum time for making an HTTP Request or fetching a Kubernetes object.
+	// Max value for conformant implementation: None
```

Comment:
There is a related value here that I expected to be configurable. I think this may be the only one that truly needs a "Max value for conformant implementation", and we should probably start with the current default of 30 seconds: https://github.com/kubernetes-sigs/gateway-api/blob/361a156b0705bfbf10accc01b8c3e7424acfbc7e/conformance/utils/http/http.go#L69

Reply:
What would you suggest as the max value for this?

Reply:
I think 30 seconds is fine as both default and what we say in the "Max value for conformant implementation" comment.

Reply:
Done.